diff --git a/app/src/behaviors/behavior_input_two_axis.c b/app/src/behaviors/behavior_input_two_axis.c
index ea8e948d..3fc12516 100644
--- a/app/src/behaviors/behavior_input_two_axis.c
+++ b/app/src/behaviors/behavior_input_two_axis.c
@@ -40,6 +40,7 @@ struct movement_state_2d {
 struct behavior_input_two_axis_data {
     struct k_work_delayable tick_work;
     const struct device *dev;
+    bool fixed_speed;
 
     struct movement_state_2d state;
 };
@@ -109,10 +110,12 @@ static inline uint8_t get_acceleration_exponent(const struct behavior_input_two_
 #endif // IS_ENABLED(CONFIG_ZMK_POINTING_SMOOTH_SCROLLING)
 
 static float speed(const struct behavior_input_two_axis_config *config, uint16_t code,
-                   float max_speed, int64_t duration_ticks) {
+                   float max_speed, int64_t duration_ticks, bool fixed_speed) {
     uint8_t accel_exp = get_acceleration_exponent(config, code);
 
-    if ((1000 * duration_ticks / CONFIG_SYS_CLOCK_TICKS_PER_SEC) > config->time_to_max_speed_ms ||
+    if (fixed_speed ||
+        (1000 * duration_ticks / CONFIG_SYS_CLOCK_TICKS_PER_SEC) >
+            config->time_to_max_speed_ms ||
         config->time_to_max_speed_ms == 0 || accel_exp == 0) {
         return max_speed;
     }
@@ -135,7 +138,7 @@ static void track_remainder(float *move, float *remainder) {
 }
 
 static float update_movement_1d(const struct behavior_input_two_axis_config *config, uint16_t code,
-                                struct movement_state_1d *state, int64_t now) {
+                                struct movement_state_1d *state, int64_t now, bool fixed_speed) {
     float move = 0;
     if (state->speed == 0) {
         state->remainder = 0;
@@ -143,10 +146,11 @@ static float update_movement_1d(const struct behavior_input_two_axis_config *con
     }
 
     int64_t move_duration = ticks_since_start(state->start_time, now, config->delay_ms);
-    LOG_DBG("Calculated speed: %f", speed(config, code, state->speed, move_duration));
+    LOG_DBG("Calculated speed: %f", speed(config, code, state->speed, move_duration, fixed_speed));
     move =
         (move_duration > 0)
-            ? (speed(config, code, state->speed, move_duration) * config->trigger_period_ms / 1000)
+            ? (speed(config, code, state->speed, move_duration, fixed_speed) *
+               config->trigger_period_ms / 1000)
             : 0;
 
     track_remainder(&(move), &(state->remainder));
@@ -154,12 +158,13 @@ static float update_movement_1d(const struct behavior_input_two_axis_config *con
     return move;
 }
 static struct vector2d update_movement_2d(const struct behavior_input_two_axis_config *config,
-                                          struct movement_state_2d *state, int64_t now) {
+                                          struct movement_state_2d *state, int64_t now,
+                                          bool fixed_speed) {
     struct vector2d move = {0};
 
     move = (struct vector2d){
-        .x = update_movement_1d(config, config->x_code, &state->x, now),
-        .y = update_movement_1d(config, config->y_code, &state->y, now),
+        .x = update_movement_1d(config, config->x_code, &state->x, now, fixed_speed),
+        .y = update_movement_1d(config, config->y_code, &state->y, now, fixed_speed),
     };
 
     return move;
@@ -187,7 +192,7 @@ static void tick_work_cb(struct k_work *work) {
     // LOG_INF("x start: %llu, y start: %llu, current timestamp: %llu", data->state.x.start_time,
     //         data->state.y.start_time, timestamp);
 
-    struct vector2d move = update_movement_2d(cfg, &data->state, timestamp);
+    struct vector2d move = update_movement_2d(cfg, &data->state, timestamp, data->fixed_speed);
 
     int ret = 0;
     bool have_x = is_non_zero_1d_movement(move.x);
@@ -247,6 +252,13 @@ int behavior_input_two_axis_adjust_speed(const struct device *dev, int16_t dx, i
     return 0;
 }
 
+int behavior_input_two_axis_set_fixed_speed(const struct device *dev, bool fixed_speed) {
+    struct behavior_input_two_axis_data *data = dev->data;
+
+    data->fixed_speed = fixed_speed;
+    return 0;
+}
+
 static int behavior_input_two_axis_init(const struct device *dev) {
     struct behavior_input_two_axis_data *data = dev->data;
 
